build/Dockerfile

```diff
@@ -1,27 +1,42 @@
-# docker build . --rm --build-arg version=2.0.4 --build-arg commit=master -t heroiclabs/nakama:2.0.4
+## Copyright 2018 The Nakama Authors
+##
+## Licensed under the Apache License, Version 2.0 (the "License");
+## you may not use this file except in compliance with the License.
+## You may obtain a copy of the License at
+##
+## http:##www.apache.org/licenses/LICENSE-2.0
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+
+# docker build "$PWD" --build-arg commit=d79ccd1 --build-arg version=v2.1.0 -t heroiclabs/nakama:2.1.0
 
 FROM golang:1.11.1-alpine3.7 as builder
 
 ARG commit
+ARG version
+
+ENV GOOS linux
+ENV GOARCH amd64
+ENV CGO_ENABLED 1
 
 RUN apk --no-cache add ca-certificates gcc musl-dev git
 
 WORKDIR /go/src/github.com/heroiclabs/
 RUN git config --global advice.detachedHead false && \
-  git clone -q -n https://github.com/heroiclabs/nakama
+  git clone -q -n https://github.com/heroiclabs/nakama nakama
 
 WORKDIR /go/src/github.com/heroiclabs/nakama
 RUN git checkout -q "$commit" && \
-  GOOS=linux GOARCH=amd64 go build && \
-  mkdir -p /go/build && \
-  mv nakama /go/build
+  go build -o /go/build/nakama -gcflags "-trimpath $PWD" -asmflags "-trimpath $PWD" -ldflags "-s -w -X main.version=$version -X main.commitID=$commit"
 
 FROM alpine:3.7
 
 MAINTAINER Heroic Labs <support@heroiclabs.com>
 
-ARG version
```

Comment:
I think we might still need this here for docker to pass the `version` arg down to this container?